### DISCUSSION TITLE
Implementation DAC PT8211

### DISCUSSION THREAD

**Discussion by compactflash** [schreibfaul1/ESP32-audioI2S#108] [Ideas]
Hello Wolle

Is there a way for you to implement a function to use the DAC PT8211 . The serial bus input data format of PT8211 is Japanese or called LSBJ (Least Significant Bit Justified) format. So it is required to change a value in your Audio.cpp  from MSB to LSB 

m_i2s_config.channel_format       = I2S_CHANNEL_FMT_RIGHT_LEFT;
    m_i2s_config.communication_format = (i2s_comm_format_t)(I2S_COMM_FORMAT_I2S | I2S_COMM_FORMAT_I2S_MSB);
    m_i2s_config.intr_alloc_flags     = ESP_INTR_FLAG_LEVEL1; // high interrupt priority

Last fall I have released a PCB for the great ESPINO32 Project with a PCM5102B. Now I am planing a new one with easier to handle parts , especially the DAC . The PT8211 is working fine and simple to solder . There is no problem for me to change it in the Audio.cpp but it would be nice to configure it in a special settings.h for my project .

**Comment by schreibfaul1** [comment]
Hi compactflash,
in setup():   with setI2SCommFMT_LSB(true) the I2S format can be changed from MSB to LSB. I don't know if that works, I don't have a PT8211 to test.

**Reply by compactflash** [reply]
Hi Wolle
I have tested it , it works fine .
Thank you very much

**Comment by hkp792** [comment]
SCHEMATIC PT8211 AND STM32F407VETX
